src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1

```diff
@@ -401,21 +401,119 @@ class Organization: ADOSVTBase
+                        }
+                        
+                        # Display extensions with Top Publishers, extensions that are private and Nonprod extensions
+                        $topPublisherExt=@()
+                        $privateExtensions=@()
+                        $nonProdExtensions=@()
+                        $extensionList | ForEach-Object {
+                            $url="https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery?api-version=6.1-preview.1"
+                            $inputbody = "{
+                                'assetTypes': null,
+                                'filters': [
+                                    {
+                                        'criteria': [
+                                            {
+                                                'filterType': 7,
+                                                'value': '$($_.publisherId).$($_.extensionId)'
+                                            }
+                                        ]                                
+                                    }
+                                ]
+                            }" | ConvertFrom-Json | ConvertTo-json -Depth 10
+    
+                            $response= Invoke-WebRequest -Uri $url `
+                                -Method Post `
+                                -ContentType "application/json" `
+                                -Body $inputbody `
+                                -UseBasicParsing
+    
+                            $responseObject=$response.Content | ConvertFrom-Json
+
+                            if([Helpers]::CheckMember($responseobject.results[0], "extensions") -eq $false )
+                            {
+                                $privateExtensions+=$_
+                            }
+                            else{
+                                if($responseobject.results.extensions.flags -match "preview")
+                                {
+                                    $nonProdExtensions+=$_
+                                }
+                                if($responseobject.results.extensions.publisher.flags -match "certified")
+                                {
+                                    $topPublisherExt+=$_
+                                }
+                            }                            
+                        }
+
+                        $controlResult.AddMessage("`nNo. of installed extensions with Top Publishers: "+$topPublisherExt.count);
+                        $controlResult.AddMessage($topPublisherExt);
+                        $controlResult.AddMessage("`nNo. of installed extensions with private visibility: "+$privateExtensions.count);
+                        $controlResult.AddMessage($privateExtensions);
+                        $controlResult.AddMessage("`nNo. of installed extensions that are non Prod: "+$nonProdExtensions.count);
```

Comment:
that belong to non production environment

Reply:
updated the message